step/src/main/java/com/google/sps/CalendarManager.java

```diff
@@ -0,0 +1,159 @@
+   *
+   * @param hostUser the owner of the Google Calendar event.
+   * @param guestUser the user to be invited to the Google Calendar event.
+   * @param event the event to be pushed to Google Calendar. This event can be conveniently
+   *              built using CalendarManager.MatchEventBuilder.
+   */
+  private static void pushMatchEvent(User hostUser, User guestUser, Event event) {
+    Calendar calendar = getCalendar(hostUser);
+    try {
+      String calendarId = "primary";
+      event = calendar.events().insert(calendarId, event).execute();
+      System.out.printf("Event created: %s\n", event.getHtmlLink());
+    } catch (IOException e) {
+      System.out.println("ERROR: " + e.getMessage());
+      System.out.println("Unable to push match event.");
+    }
+  }
+
+  private static Calendar getCalendar(User user) {
+    return new Calendar.Builder(
+        new NetHttpTransport(),
+        new JacksonFactory(),
+        user.getCredential()
+      ).setApplicationName("Friend Matching Plus").build();
+  }
+
+}
+
+class MatchEventBuilder {
+  private Event matchEvent;
+  private static final int DEFAULT_EVENT_LENGTH_MINUTES = 60;
+
+  public MatchEventBuilder() {
+    matchEvent = new Event();
+  }
+
+  public MatchEventBuilder setAttendees(User hostUser, User guestUser) {
+    String eventSummary = String.format("FMP: %s / %s", hostUser.getName(), guestUser.getName());
+    matchEvent.setSummary(eventSummary);
+
+    String eventDescription = String.format("%s and %s have matched on Friend Matching Plus! ",
+                                              hostUser.getName(), guestUser.getName());
+    eventDescription += "You can join a video call with your match by clicking "
+                        + "\"Join with Google Meet\"";
+    matchEvent.setDescription(eventDescription);
+
+    matchEvent.setAttendees(Arrays.asList(new EventAttendee[] {
+      new EventAttendee().setEmail(guestUser.getEmail())
+    }));
+
+    return this;
+  }
+
+  // ZonedDateTime guide: https://www.baeldung.com/java-8-date-time-intro
+  public MatchEventBuilder setStartDateTime(int year, int month, int day, int hour, int minute) {
+    String dateTimeString = dateTimeString(year, month, day, hour, minute);
+
+    EventDateTime start = new EventDateTime()
+      .setDateTime(new DateTime(dateTimeString));
+      // .setTimeZone("America/Los_Angeles"); // TODO how to configure time zones?
+    matchEvent.setStart(start);
+
+    // Assume a 1 hour event.
+    DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateTimeString, dateTimeFormatter);
+    zonedDateTime = zonedDateTime.plusMinutes(DEFAULT_EVENT_LENGTH_MINUTES);
+    dateTimeString = zonedDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+
+    EventDateTime end = new EventDateTime()
+      .setDateTime(new DateTime(dateTimeString));
+    matchEvent.setEnd(end);
+
+    return this;
```

Comment:
Set method often returns null. If you want to get the result, then you can create a get method returning MatchEventBuilder

Reply:
None of these methods return null, they return the current MatchEventBuilder object so it can be used as a [Builder](https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/) object and allows for the simple syntax found in createMatchEvent(). The builder pattern doesn't really support get methods but supports a build() method that is used to signal the end of the build pattern so the builder class can return the actual object it was building.